src/engine.ts

```diff
@@ -29,6 +37,13 @@ export default class Engine {
   roundBoosters:  {
     [key in Booster]?: boolean 
   } = { }; 
+  techTiles: {
+    [key in TechTilePos]?: {tile: TechTile; numTiles: number}
+  } = {};
+  advTechTiles: {
```

Comment:
Actually, reading the rules again, there are only 6 advanced tech tiles available, one for each research track. 

As such it can be `[key in AdvTechTilePos]?: AdvTechTile`, set to `undefined` when taken (or directly `delete this.advTechTiles[posOfTileTaken]`).